### PR TITLE
Update bare metal deployment code to use semantic version

### PR DESCRIPTION
**Pull request by prsurve**
Signed-off-by: prsurve <prsurve@redhat.com>

### CODE REVIEW

**Review by ebenahar — APPROVED** [review]

**Comment by prsurve** [comment]
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/8730/console

**Review by OdedViner — APPROVED** [review]

File: ocs_ci/deployment/baremetal.py [hunk 1]
```diff
@@ -290,13 +290,12 @@ def deploy(self, log_cli_level="DEBUG"):
...
             logger.info("Uploading PXE files")
             ocp_version = get_ocp_version()
-            float_ocp_version = float(ocp_version)
             for machine in self.mgmt_details:
                 if self.mgmt_details[machine].get("cluster_name") or self.mgmt_details[
                     machine
                 ].get("extra_node"):
                     pxe_file_path = self.create_pxe_files(
-                        ocp_version=float_ocp_version,
+                        ocp_version=ocp_version,
```
Comment (Avilir) [inline ocs_ci/deployment/baremetal.py:298]: should it be `Version.coerce(ocp_version)` ?

Comment (prsurve) [reply ocs_ci/deployment/baremetal.py:298]: no not needed here doing it here https://github.com/red-hat-storage/ocs-ci/pull/5279/files#diff-05053122d6e5936ca528fcd1643379f792a8350f094e1c0aa5c62903f86a8746R579

**Review by Avilir — APPROVED** [review]

File: ocs_ci/deployment/baremetal.py [hunk 2]
```diff
@@ -577,11 +576,11 @@ def create_pxe_files(self, ocp_version, role):
             extra_data = ""
             bm_install_files_loc = self.helper_node_details["bm_install_files"]
             extra_data_pxe = "rhcos-live-rootfs.x86_64.img coreos.inst.insecure"
-            if ocp_version <= 4.6:
+            if Version.coerce(ocp_version) <= Version.coerce("4.6"):
```
Comment (petr-balogh) [inline ocs_ci/deployment/baremetal.py:579]: Hey @prsurve ,

Can you please use approach from version module I added?
See examples here:
https://github.com/red-hat-storage/ocs-ci/pull/4872/files

See email:
https://mailman-int.corp.redhat.com/archives/ocs-qe/2021-September/msg00292.html

**Pull request merged**